File: app/src/main/java/com/hades/example/android/qa/AlertParamGetter3.java
package com.hades.example.android.qa;

import android.app.Activity;

import com.hades.example.android.R;
import com.hades.example.android.app_component._activity._life_cycle.A;

class AlertParamGetter3 {
    private Activity context;
    private DefaultUIConfigure configure;

    public AlertParamGetter3(Activity context, DefaultUIConfigure configure) {
        this.context = context;
        this.configure = configure;
    }

    public AlertParam getAlertParam() {
        AlertParam param = getDefaultAlertParam();
        setCustomAlertParam(param, configure);
        return param;
    }

    public AlertParam getDefaultAlertParam() {
        AlertParam param = new AlertParam();
        param.setTitle(getDefaultTitle());
        return param;
    }

    public void setCustomAlertParam(AlertParam param, DefaultUIConfigure configure) {
        if (null == configure) {
            return;
        }
        if (null == configure.getDefaultAlertConfigure()) {
            return;
        }
        if (null != configure.getDefaultAlertConfigure().getTitle()) {
            param.setTitle(configure.getDefaultAlertConfigure().getTitle());
        }
    }

    public String getDefaultTitle() {
        return context.getString(R.string.default_title);
    }
}

File: app/src/main/java/com/hades/example/android/qa/v2/AlertParamGetter3.kt
package com.hades.example.android.qa.v2

import android.app.Activity
import com.hades.example.android.R
import com.hades.example.android.qa.AlertParam
import com.hades.example.android.qa.DefaultUIConfigure

internal open class AlertParamGetter3(
    private val context: Activity,
    private val configure: DefaultUIConfigure?
) {
    internal fun getAlertParam(): AlertParam {
        val param = getDefaultAlertParam()
        setCustomAlertParam(param)
        return param
    }

    private fun getDefaultAlertParam(): AlertParam {
        val param = AlertParam()
        param.title = getDefaultTitle()
        return param
    }


    private fun setCustomAlertParam(param: AlertParam) {
        if (null == configure) {
            return
        }
        if (null == configure.defaultAlertConfigure) {
            return
        }
        if (null != configure.defaultAlertConfigure!!.title) {
            param.title = configure.defaultAlertConfigure!!.title
        }
    }

    open fun getDefaultTitle(): String? {
        return context.getString(R.string.default_title)
    }
}
